### PR TITLE
Move mempool, orphanage, API tests to new structure

### PR DESCRIPTION
**Pull request by geigerzaehler**
In three separate commits we
* move `Oscoin.Test.API` to `Test.Oscoin.API`
* move `Oscoin.Test.Storage.Block.Orphanage` to `Test.Oscoin.Storage.Block.Orphanage`
* extract `Test.Oscoin.Node.Mempool` from `Oscoin.Tests`

### CODE REVIEW

File: test/Test/Oscoin/API.hs [hunk 1]
```diff
@@ -24,8 +24,8 @@ import           Test.Tasty.HUnit.Extended
 import           Test.Tasty.QuickCheck
 
 
-tests :: forall c. Dict (IsCrypto c) -> [TestTree]
-tests Dict =
+tests :: forall c. Dict (IsCrypto c) -> TestTree
+tests Dict = testGroup "Oscoin.API"
```
Comment (kim) [inline test/Test/Oscoin/API.hs]: Wait .. shouldn't this always be _literally_ the module's name?

Comment (geigerzaehler) [reply test/Test/Oscoin/API.hs]: I’d be ok with that, too. But I omitted the `Test` prefix in all my changes and this is also what is documented in [“Test organization”](https://github.com/oscoin/oscoin#test-organization) in the readme.

Comment (kim) [reply test/Test/Oscoin/API.hs]: Ok, guess I missed that. I think mechanically the module name is preferable, as we can eventually just have it generated.

Comment (geigerzaehler) [reply test/Test/Oscoin/API.hs]: I’ll switch to that

**Comment by geigerzaehler** [comment]
Can you have another look @kim?

**Pull request merged**